src/devices/MotorHat/MotorHat.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.Pwm;
+using System;
+using System.Device.I2c;
+using System.Collections.Generic;
+using MotorHat;
+using System.Device.Pwm;
+
+namespace Iot.Device.MotorHat
+{
+    /// <summary>
+    /// Raspberry Pi Motor Hat based on PCA9685 PWM controller
+    /// </summary>
+    public class MotorHat : IDisposable
+    {
+        /// <summary>
+        /// Default I2C address of Motor Hat
+        /// </summary>
+        public const int I2cAddressBase = 0x60;
+
+        /// <summary>
+        /// Motor Hat is built on top of a PCa9685
+        /// </summary>
+        private readonly Pca9685 pca9685;
+
+        /// <summary>
+        /// Holds every channel that is being used by either a DCMotor, Stepper, ServoMotor, or PWM
+        /// </summary>
+        private IList<PwmChannel> channelsUsed = new List<PwmChannel>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MotorHat"/> class with the specified I2C settings and PWM frequency.
+        /// </summary>
+        /// <param name="settings">The I2C settings of the MotorHat.</param>
+        /// <param name="frequency">The frequency in Hz to set the PWM controller.</param>
+        /// <remarks>
+        /// The default i2c address is 0x60, but the HAT can be configured in hardware to any address from 0x60 to 0x7f.
+        /// The PWM hardware used by this HAT is a PCA9685. It has a total possible frequency range of 24 to 1526 Hz.
+        /// Setting the frequency above or below this range will cause PWM hardware to be set at its maximum or minimum setting.
+        /// </remarks>
+        public MotorHat(I2cConnectionSettings settings, double frequency)
```

Comment:
use default value for frequency, this way you can get rid of half of the overloads